### ISSUE TITLE
!fileatask

### ISSUE THREAD

**Issue by abizer** [ocf/ircbot#77]
next in the line of ircbot functionality to steal from Facebook, we should have a plugin that files GitHub issues against the mentioned repository using the GH API, e.g.

`!fileatask ircbot make a fileatask plugin`
`!fileatask ocfweb migrate to kubernetes`

**Comment by jvperrin** [comment]
Also `@marley ticket me [title] [description] [project]`

Lol

**Comment by abizer** [comment]
nice - we need more diversity in command characters too. We only use `!` and `#` so far, imagine the possibilities with `$`, `&`, `@`, `%`, `∆`, `.`, `|`, `:`, etc etc

**Comment by chriskuehl** [comment]
On the other hand, in my experience tickets filed via IRC bot tend to be pretty low-quality (very short on context, which is a problem with a lot of our tickets already). Lots of these tickets end up getting closed months or years later because nobody really remembers what they refer to. Most tickets really do benefit from somebody typing a few sentences into a textbox in the GitHub UI.

I guess it might be convenient for smaller tasks where the ticket is basically serving as a reminder for a specific person.

**Comment by abizer** [comment]
yeah, that's mostly the use case I envision. little bug reports, minor feature requests, or reminders. if we get our own moffle set up (@keur) we could also have the bot include a link to the context to help improve the quality aspect.

**Comment by dkess** [comment]
Yup, if we were to do this I'd definitely want it to link back to the discussion so we can get good context.

There are other difficulties with this though. We don't have a centralized task management system, so we'd either have to get one (probably a bad idea) or would have to get this to post to a specific GitHub repo (which would make the syntax more annoying).

I'd be willing to entertain more people's ideas on this, but I'm still unconvinced that this could be done well.

**Comment by jvperrin** [comment]
Well we have rt, which I'd call our (somewhat) centralized task management system. The main issue I think we have is that right now we use any and all of (1) GitHub issues, (2) RT, (3) Discourse for tracking things to do, so supporting all of those would be tougher and confusing.

Maybe if we can have it open an issue in any GitHub repo that would be enough?

**Comment by abizer** [comment]
I mean, the second argument in the given examples is the repository to file
tasks against (ircbot, ocfweb), so I don't think we need to worry about any
of this?
